pkg/kubernetes/reconcile.go

```diff
@@ -14,138 +14,129 @@ import (
 	"github.com/grafana/tanka/pkg/spec/v1alpha1"
 )
 
-// Reconcile extracts all valid Kubernetes objects from the raw output of the
-// Jsonnet compiler. A valid object is identified by the presence of `kind` and
-// `apiVersion`.
-// TODO: Check on `metadata.name` as well and assert that they are
-// not only set but also strings.
-func Reconcile(raw map[string]interface{}, spec v1alpha1.Spec, targets []*regexp.Regexp) (state manifest.List, err error) {
-	extracted, err := extract(raw)
+// Manifest is a type alias so we can use local variable name manifest without colliding
+// with the package
+type Manifest = manifest.Manifest
```

Comment:
Please avoid aliasing types for cosmetic purposes. I don't think it improves code quality a lot and it can be confusing down the road if you wonder what's the difference between `pkg/kubernetes.Manifest` and `pkg/kubernetes/manifest.Manifest`.

Furthermore, these types are not completely equal anymore, Go differentiates between them in weird occasions.

Please stick to `manifest.Manifest` for `map[string]interface{}` and `manifest.List` for `[]map[string]interface{}`

Reply:
> Furthermore, these types are not completely equal anymore, Go differentiates between them in weird occasions.

That's interesting, I wasn't aware of this. I was under the impression that type aliasing was transparent, whereas type definition `type Manifest manifest.Manifest` would be different.

I'm happy to change it back, but I was curious what situation Go differentiates between them?

Reply:
Oh, I missed that `=` sign there ... yes you are right, from a technical perspective these two are equal.

However, as the type alias proposal states, "motivation is to enable gradual code repair during large-scale refactorings". I don't see a super big win in having `manifest` variables compared to the fact that we are shadowing the package. This is confusing to read, as we are using `manifest` for the package throughout the rest of the codebase.